docs/proposal/network-topology.md

```diff
@@ -0,0 +1,215 @@
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+At this stage, Volcano already supports task-level network topology.
+
+```yaml
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: PodGroup
+metadata:
+  name: network-topology-podgroup
+spec:
+  minMember: 6
+  networkTopology:
+    mode: hard 
+    highestTierAllowed: 2
+  subGroupPolicy: 
+    - subGroupSize: 3
+      minSubGroups: 2
+      name: task
+      matchPolicy:
+        - labelKey: volcano.sh/task-subgroup-id
+      networkTopology:
+        mode: hard 
+        highestTierAllowed: 1
+```
+
+A `subGroupPolicy` has been added to the `podGroup` to ensure task-level gang scheduling and network topology.
+
+- `subGroupSize`: The number of pods in a subGroup.
+- `minSubGroups`: The minimum replicas of subGroups.
+- `matchPolicy`: The label key used to match pods.
+- `networkTopology`: The network topology of a subGroup.
+
+`ModelServing` provides a `GangPolicy`, which determines the minimum number of replicas expected for a role. Based on this configuration, we can create `podGroup`.
+
+```yaml
+# gangPolicy
+gangPolicy:
+  minRoleReplicas:
+    prefill: 2    # Require at least 2 prefill role replicas
+    decode: 1     # Require at least 1 decode role replica
+
+# PodGroup
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: PodGroup
+metadata:
+  name: network-topology-podgroup
+spec:
+  networkTopology:
+    mode: hard 
+    highestTierAllowed: 2
+  minTaskMember:
+    "prefill-0": 4    # 1 entry + 3 workers for prefill role replica 0
+    "prefill-1": 4    # 1 entry + 3 workers for prefill role replica 1
+    "decode-0": 3     # 1 entry + 2 workers for decode role replica 0
+    # ... additional role replicas
+  minResources:
+  subGroupPolicy: 
+    - name: network-topology
+      matchPolicy:
+        - labelKey: "modelserving.volcano.sh/role"
+        - labelkey: "modelserving.volcano.sh/role-id"
+      networkTopology:
+        mode: hard 
+        highestTierAllowed: 1
+```
+
+Using minTaskMember, specify the number of pods required for 2P instances and 1D instance. When creating a pod, ModelServing will add some labels to it. For example, `modelServing controller` creates a prefill-0 pod, this pod will add the following labels:
+
+```yaml
+modelserving.volcano.sh/role: prefill
+modelserving.volcano.sh/role-id: prefill-0
+```
+
+So we can use the labels "modelserving.volcano.sh/role" and "modelserving.volcano.sh/role-id" to group the pods that need to be deployed.
+
+Therefore, we still provide the `NetworkTopology` within the `ServingGroup` to ensure that the overall `NetworkTopology` of the `ServingGroup` aligns with the `NetworkTopology` of the `Roles`.
+
+Therefore, we still provide the `NetworkTopologySpec` within the `ServingGroup`, where you can configure both the ServingGroup-level network topology and the role-level network topology.
```

Comment:
This is the keyPoint, how does you place `NetworkTopologySpec` in our modelServing, should explicitly document it, and better give a demo